particle-iot/device-os-protobuf
# Introduce deviceOSProtobuf.getDefinition(name) to device-os-protobuf

## Overview

Introduces main public API methods for this module:

- `encode(protobufMessageName, {data: "foo"})`
- `decode(protobufMessageName, buffer)`
- `getDefinition(protobufMessageName)`
- `getDefinitions()` / `getNamespaces()`

Story details: https://app.shortcut.com/particle/story/95530

This work unblocks https://app.shortcut.com/particle/story/95530/introduce-deviceosprotobuf-getdefinition-name-to-device-os-protobuf#activity-95768

which in turn unblocks: https://app.shortcut.com/particle/story/94987/add-usb-validate-fqc-commands-to-mfg-cli

Hoping we can merge this without too much tussle/yak-shaving so i can get back to the P2 mfg-cli motivation behind this work before too long.

## How to test?

1. `git checkout sc-95530/introduce-getDefinition-function`
2. `npm i`
3. `npm run test:ci`; observe linter, tests, and coverage stats.
4. Read the README.md; note anywhere things could be written more clearly.
5. Read the rendered `docs/reference.md`; ask questions if anything isn't clear.
